core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -207,6 +267,18 @@ private[spark] object ChunkedByteBuffer {
     }
     out.toChunkedByteBuffer
   }
+
+  /**
+   * Try to estimate appropriate chunk size so that it's not too large(waste memory) or too
+   * small(too many segments)
+   */
+  def estimateBufferChunkSize(estimatedSize: Long = -1): Int = {
+    if (estimatedSize < 0) {
+      CHUNK_BUFFER_SIZE
+    } else {
+      Math.max(Math.min(estimatedSize >> 3, CHUNK_BUFFER_SIZE).toInt, MINIMUM_CHUNK_BUFFER_SIZE)
```

Comment:
Why are we doing `estimatedSize >> 3` ? Why not leave it as is ?

Reply:
The logic is explained in the [below comments](https://github.com/apache/spark/pull/38064#discussion_r999794248), the heuristic is somehow arbitrary but I think it should make some sense. Feel free to leave any feedback, thanks!

Reply:
I don't think it is even worth it. You will only select a different from CHUNK_BUFFER_SIZE size when it is between CHUNK_BUFFER_SIZE and MINIMUM_CHUNK_BUFFER_SIZE and in that case you can choose either anyway.

Reply:
I would focus more on reasonable estimation and preventing orders of magnitude wastage.
The cost between a 8k and 10k buffer, in this context, is negligible if not zero.

Reply:
Changed to `estimatedSize`, that makes sense and should be good in this case. Thanks for the suggestions!